core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -5904,6 +5919,320 @@ public void testJsonArrayInAggregationContext()
+                .hasMessage("line 1:57: Invalid descriptor argument SCHEMA. Descriptors should be formatted as 'DESCRIPTOR(name [type], ...)'");
+
+        assertFails("SELECT * FROM TABLE(system.descriptor_argument_function(schema => 1))")
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessage("line 1:57: Invalid argument SCHEMA. Expected descriptor, got expression");
+
+        assertFails("SELECT * FROM TABLE(system.descriptor_argument_function(schema => TABLE(t1)))")
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessage("line 1:57: Invalid argument SCHEMA. Expected descriptor, got table");
+
+        assertFails("SELECT * FROM TABLE(system.descriptor_argument_function(schema => DESCRIPTOR(x verybigint)))")
+                .hasErrorCode(TYPE_MISMATCH)
+                .hasMessage("line 1:80: Unknown type: verybigint");
+    }
+
+    @Test
+    public void testScalarArgument()
+    {
+        analyze("SELECT * FROM TABLE(system.two_arguments_function('foo', 1))");
+
+        assertFails("SELECT * FROM TABLE(system.two_arguments_function(text => 'a', number => DESCRIPTOR(x integer, y boolean)))")
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessage("line 1:64: Invalid argument NUMBER. Expected expression, got descriptor");
+
+        assertFails("SELECT * FROM TABLE(system.two_arguments_function(text => 'a', number => DESCRIPTOR(1 + 2)))")
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessage("line 1:64: 'descriptor' function is not allowed as a table function argument");
+
+        assertFails("SELECT * FROM TABLE(system.two_arguments_function(text => 'a', number => TABLE(t1)))")
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessage("line 1:64: Invalid argument NUMBER. Expected expression, got table");
+
+        assertFails("SELECT * FROM TABLE(system.two_arguments_function(text => 'a', number => (SELECT 1)))")
+                .hasErrorCode(EXPRESSION_NOT_CONSTANT)
+                .hasMessage("line 1:74: Constant expression cannot contain a subquery");
+    }
+
+    @Test
+    public void testCopartitioning()
+    {
+        // TABLE(t1) is matched by fully qualified name: tpch.s1.t1. It matches the second copartition item s1.t1.
+        // Aliased relation TABLE(SELECT 1, 2) t1(x, y) is matched by unqualified name. It matches the first copartition item t1.
+        analyze("""
+                      SELECT * FROM TABLE(system.two_table_arguments_function(
+                                                                               input1 => TABLE(t1) PARTITION BY (a, b),
+                                                                               input2 => TABLE(SELECT 1, 2) t1(x, y) PARTITION BY (x, y)
+                                                                               COPARTITION (t1, s1.t1)))
+                """);
```

Comment:
There's a lot of unnecessary space:

```java
analyze("""
        SELECT * FROM TABLE(system.two_table_arguments_function(
            input1 => TABLE(t1) PARTITION BY (a, b),
            input2 => TABLE(SELECT 1, 2) t1(x, y) PARTITION BY (x, y)
            COPARTITION (t1, s1.t1)))
        """);
```

Note that the contents of the text block should be lined up with the the closing `"""`, otherwise the string will contain leading space at the beginning of each line:

For example:

```java
System.out.println("""
        a
        """);
```

prints:

```
a
```

while:


```java
System.out.println("""
        a
    """);
```

prints:

```
    a
```